src/lib.rs

```diff
@@ -1011,6 +1011,144 @@ where
     fn resize_stamp(n: usize) -> isize {
         n.leading_zeros() as isize | (1 << (RESIZE_STAMP_BITS - 1)) as isize
     }
+
+    /// Removes the key (and its corresponding value) from this map.
+    /// This method does nothing if the key is not in the map.
+    /// Returns the previous value associated with the given key.
+    pub fn remove<'g>(&'g self, key: &K, guard: &'g Guard) -> Option<&'g V> {
+        let h = self.hash(key);
+
+        let mut table = self.table.load(Ordering::SeqCst, guard);
+
+        loop {
+            if table.is_null() {
+                break;
+            }
+
+            // safety: table is a valid pointer.
+            //
+            // we are in one of two cases:
+            //
+            //  1. if table is the one we read before the loop, then we read it while holding the
+            //     guard, so it won't be dropped until after we drop that guard b/c the drop logic
+            //     only queues a drop for the next epoch after removing the table.
+            //
+            //  2. if table is set by a Moved node (below) through help_transfer, it will use the
+            //     `next_table` raw pointer from inside the Moved. how do we know that that is safe?
+            //
+            //     we must demonstrate that if a Moved(t) is _read_, then t must still be valid.
+            //     FIXME cf. put
+            let t = unsafe { table.deref() };
+            let n = t.bins.len() as u64;
+            if n == 0 {
+                break;
+            }
+            let i = ((n - 1) & h) as usize;
```

Comment:
This should probably use `Table::bin`. I think in general it'd be good to model this function as closely as possible to what is in [the ported code for `put`](https://github.com/jonhoo/flurry/blob/437663f284aa68d01cd806cbbd0a1b2bc7ce8ea9/src/lib.rs#L476) so that we keep the things looking similar.